app/models/input_template_renderer.rb

```diff
@@ -40,7 +49,49 @@ def input(name)
     end
   end
 
+  def render_template(template_name, input_values = {}, options = {})
+    options.assert_valid_keys(:with_foreign_input_set)
```

Comment:
It's not a Rails, but still would be doable, just would need to not pass the additional options to overridden method.